node/core/approval-voting/src/lib.rs

```diff
@@ -1802,39 +1919,65 @@ where
 			)),
 	};
 
-	let (claimed_core_index, assigned_candidate_hash) =
-		match block_entry.candidate(candidate_index as usize) {
-			Some((c, h)) => (*c, *h),
+	let n_cores = session_info.n_cores as usize;
+
+	// Early check the candidate bitfield and core bitfields lengths < `n_cores`.
+	// `approval-distribution` already checks for core and claimed candidate bitfields
+	// to be equal in size. A check for claimed candidate bitfields should be enough here.
+	if candidate_indices.len() >= n_cores {
```

Comment:
Shouldn't this be like this ?

```suggestion
	if candidate_indices.len() > n_cores {
```